nixos/modules/services/misc/nzbget.nix

```diff
@@ -61,10 +61,14 @@ in {
         p7zip
       ];
       preStart = ''
+        if [ ! -d ${cfg.dataDir} ]; then
+          echo "${cfg.dataDir} directory not found. Creating it."
+          install -d -o ${cfg.user} -g ${cfg.group} ${cfg.dataDir}
```

Comment:
Use `systemd.tmpfiles` instead for doing this.

Reply:
Could you expand a little on why `systemd.tmpfiles` might be a better approach here?  
The resulting directory is intended to be "long-lasting".

Reply:
I guess the name `tmpfiles` is a bit misleading. It can be used for non-temporary files as well, which is in fact the main use of it in NixOS. Because this approach is declarative, it is preferred in NixOS.

Reply:
Ah okay, great. Thanks for the suggestion, I'll take a look!

Reply:
Oh and actually, there's `serviceConfig.StateDirectory` too, which is preferred over tmpfiles if it works (it's not as flexible as tmpfiles, but I don't think that flexibility is needed for this specific case).